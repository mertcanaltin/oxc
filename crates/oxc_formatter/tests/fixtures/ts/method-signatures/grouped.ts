
type A = {
  new(...args): T<{
    A
  }
  >
};


type A1 = {
  (...args): T<{
    A
  }
  >
};

type A2 = {
  bar(
    ...args
  ): T<{
    A
  }>
}

class A3 {
  constructor(
    public eventName: string,
    data?: object,
  ) { }
}

class A4 {
  publicLog<
    E extends ClassifiedEvent<OmitMetadata<T>>,
    T extends IGDPRProperty,
  >(
    eventName: string, data?: object
  ) {
  }
}

const A5 = {
  publicLog<
    E extends ClassifiedEvent<OmitMetadata<T>>,
    T extends IGDPRProperty,
  >(
    eventName: string,
    data?: object,
  ) { }
}
