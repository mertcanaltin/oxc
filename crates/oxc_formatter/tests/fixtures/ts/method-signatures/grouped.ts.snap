---
source: crates/oxc_formatter/tests/fixtures/mod.rs
---
==================== Input ====================

type A = {
  new(...args): T<{
    A
  }
  >
};


type A1 = {
  (...args): T<{
    A
  }
  >
};

type A2 = {
  bar(
    ...args
  ): T<{
    A
  }>
}

class A3 {
  constructor(
    public eventName: string,
    data?: object,
  ) { }
}

class A4 {
  publicLog<
    E extends ClassifiedEvent<OmitMetadata<T>>,
    T extends IGDPRProperty,
  >(
    eventName: string, data?: object
  ) {
  }
}

const A5 = {
  publicLog<
    E extends ClassifiedEvent<OmitMetadata<T>>,
    T extends IGDPRProperty,
  >(
    eventName: string,
    data?: object,
  ) { }
}

==================== Output ====================
type A = {
  new (...args): T<{
    A;
  }>;
};

type A1 = {
  (...args): T<{
    A;
  }>;
};

type A2 = {
  bar(...args): T<{
    A;
  }>;
};

class A3 {
  constructor(
    public eventName: string,
    data?: object,
  ) {}
}

class A4 {
  publicLog<
    E extends ClassifiedEvent<OmitMetadata<T>>,
    T extends IGDPRProperty,
  >(eventName: string, data?: object) {}
}

const A5 = {
  publicLog<
    E extends ClassifiedEvent<OmitMetadata<T>>,
    T extends IGDPRProperty,
  >(eventName: string, data?: object) {},
};

===================== End =====================
